actions/README.md

```diff
@@ -0,0 +1,102 @@
+# Fragments Actions
+It is a tool-kit for integrating with different data sources, especially with HTTP-based APIs. It 
+implements stability patterns to prevent/handle network issues.
+
+## How to start?
+- add `knotx-fragments-action-core.X.Y.Z.jar` to your dependencies
+- configure the [action](#action), e.g. call the HTTP endpoint with the circuit breaker pattern implemented:
+  ```hocon
+  fetch-user-with-cb {
+    factory = cb
+    config {
+      circuitBreakerName = delivery-cb
+      circuitBreakerOptions {
+        maxRetries = 2
+        timeout = 500
+      }
+    }
+    doAction = fetch-user
+  }
+  
+  fetch-user {
+    factory = http
+    config.endpointOptions {
+      path = "/service/user/{params.id}"
+      domain = localhost
+      port = 1024
+      allowedRequestHeaders = ["Content-Type"]
+    }
+  }  
+  ```
+- use the [action provider](#action-provider) to and invoke the action e.g.
+  ```java
+  provider.get("fetch-user-with-cb").apply(fragmentContext, resultHandler)
+  ```
+ 
+See the [Core module documentation](https://github.com/Knotx/knotx-fragments/tree/master/action/core) 
+for a full list of [actions](#action).
+
+## How does it work?
+[Action](#action) is ready to use [fragment operation](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment-operation) 
+created by an [action factory](#action-factory). Action may be parametrized with JSON (`config`). 
+You can use actions' factories to create actions, however, we recommend using the [action provider](#action-provider) 
+that would hide the complexity of initialization (maintaining stateful actions, combining actions 
+with their behaviours). 
+
+### Action
+Action is a single [fragment operation](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment-operation) 
+that operates on a [Fragment](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment).
+Action can invoke an API and save the response body in a fragment payload or simply modify a fragment's body.
+
+Action may be decorated with [behaviours](https://github.com/Knotx/knotx-fragments/tree/master/action/core#behaviours). 
+Behaviour is a special action that wraps the original action and adds some functionality e.g 
+employ the circuit breaker pattern. 
+
+Please note that actions can be **stateless** and **stateful**.
+
+### Knot
+**Knot** is a scalable **Action** that is available on the [Vert.x Event Bus](https://vertx.io/docs/vertx-core/java/#event_bus). 
+See the [Template Engine](https://github.com/Knotx/knotx-template-engine/blob/master/core/src/main/java/io/knotx/te/core/TemplateEngineKnot.java) 
+module as reference.
+
+### Action provider
+Action provider initializes actions, combines actions with behaviours and caches stateful ones. It
+uses aliases to identify a particular action. Each alias points to action factory name, its configuration.
+If action is a behaviour, it points also to wrapped action (also by alias).
```

Comment:
```suggestion
Action provider initializes actions, combines actions with behaviours and caches stateful ones. It
uses aliases to identify a particular action. Each alias points to an action factory name and therefore its configuration.
If action is a behaviour, it also points to a wrapped action (by alias as well).
```
Don't know if `and therefore` is appropriate but I'd rephrase this sentence in some way to be more clear